src/magic_folder/test/test_web.py

```diff
@@ -570,3 +586,126 @@ def get_status(self):
 class StubMagicFolder(object):
     uploader = attr.ib(default=attr.Factory(StubQueue))
     downloader = attr.ib(default=attr.Factory(StubQueue))
+
+
+class AuthorizationTests(SyncTestCase):
+    """
+    Tests for the authorization requirements for resources beneath ``/v1``.
+    """
+    @given(
+        good_token=tokens(),
+        bad_tokens=lists(tokens()),
+        child_segments=lists(text()),
+    )
+    def test_unauthorized(self, good_token, bad_tokens, child_segments):
+        """
+        If the correct bearer token is not given in the **Authorization** header
+        of the request then the response code is UNAUTHORIZED.
+
+        :param bytes good_token: A bearer token which, when presented, should
+            authorize access to the resource.
+
+        :param bad_tokens: A list of bearer token which, when presented all at
+            once, should not authorize access to the resource.  If this is
+            empty no tokens are presented at all.  If it contains more than
+            one element then it creates a bad request with multiple
+            authorization header values.
+
+        :param [unicode] child_segments: Additional path segments to add to the
+            request path beneath **v1**.
+        """
+        # We're trying to test the *unauthorized* case.  Don't randomly hit
+        # the authorized case by mistake.
+        assume([good_token] != bad_tokens)
+
```

Comment:
Should this be something more like `assume(good_token not in bad_tokens)`..? Or is `bad_tokens` always a 1-element list?

Reply:
`bad_tokens` is any length.  The reason I did the (non-)equality test is to allow `bad_tokens` to equal things like `[good_token, bad_token]` and include those in the rejected case.  If somebody shows up with two `Authorization` headers, something weird is happening and it seems like fail-closed is the right move.  I'm open to being convinced otherwise though.

Reply:
Ahh, I see. Okay yeah that makes sense.